tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -1953,6 +2113,32 @@ uint32_t SPIRVEmitter::createVectorSplat(const Expr *scalarExpr, uint32_t size,
   }
 }
 
+void SPIRVEmitter::splitVecLastElement(QualType vecType, uint32_t vec,
+                                       uint32_t *residual,
+                                       uint32_t *lastElement) {
+  assert(hlsl::IsHLSLVecType(vecType));
+
+  const uint32_t count = hlsl::GetHLSLVecSize(vecType);
+  const uint32_t elemTypeId =
+      typeTranslator.translateType(hlsl::GetHLSLVecElementType(vecType));
+
+  if (count == 2) {
+    *residual = theBuilder.createCompositeExtract(elemTypeId, vec, 0);
+  } else {
+    assert(count > 2);
```

Comment:
no need for this assert. It will never be triggered if the above assert is added.